### PR TITLE
use option in all the add func

### PR DESCRIPTION
**Pull request by ianzhang366**
Reading our recent changes, it seems we need to do lots of replacing/modify if we change the
`AddFuncs`.

To avoid this(or make the add func signature stable), I put the `CMD option` to its own package.

Then the initialization process would be,
 There's an `options`  variable initialed in the `cmd/exec/options.go`, which is then used by 
`manager.go`. From the ` manager.go`, we just pass along the `options` variable to all the 
`AddFuncs`. At each `AddFuncs`, it will pick what it needs.

In the future, if we are going to change the options for `AddFuncs`, we just need to change the 
option structure and the specific `AddFuncs` instead of all the `AddFuncs` signatures.

### CODE REVIEW

**Review by xiangjingli — APPROVED** [review]

**Review by mikeshng — APPROVED** [review]
/lgtm 
This is an excellent refactoring @ianzhang366 !!! but please fully test with ACM before merge.
Most importantly there should only be one helmrelease controller running even with the local cluster enable.

**Review by rokej — APPROVED** [review]
If this does not fix anything, I feel this type of global change should have been made early in the dev cycle. As Mike mentioned, please make sure correct controllers are started in the right environment. Thank you.

**Comment by ianzhang366** [comment]
> If this does not fix anything, I feel this type of global change should have been made early in the dev cycle. As Mike mentioned, please make sure correct controllers are started in the right environment. Thank you.

totally agree, I will merge this one after the release then

File: pkg/controller/controller.go
Comment (mikeshng) [inline pkg/controller/controller.go:47]: @ianzhang366 @xiangjingli do you think we should change this line to
```
if ops.Standalone || !utils.IsHub(m.GetConfig()) {
```
I am not sure if the golang compiler already does this optimization automatically but its better if we just do it ourselves to make sure. I can make the change to this branch if you guys agree.

Comment (ianzhang366) [reply pkg/controller/controller.go:47]: good catch, but this PR is planning to be merged after the 2.2 release, if that timeline is ok for you, please make the change

**Pull request closed without merging**
